src/de/flavors.rs

```diff
@@ -95,6 +95,9 @@ pub trait Flavor<'de>: 'de {
     /// Obtain the next byte for deserialization
     fn pop(&mut self) -> Result<u8>;
 
+    /// Returns the number of bytes remaining in the message, if known.
```

Comment:
Do you think it makes sense to make this optional for now? e.g. `fn size_hint(&self) -> Option<usize> { None }`?

Either way, it might also be good to note that "this number might be a lie", sort of like the Iterator size hint does (e.g. "this must still work right if the size hint is wrong")

Reply:
Good points. Added.